app/services/katello/repository_type.rb

```diff
@@ -90,7 +92,9 @@ def as_json(_options = {})
         :name => self.id.to_s,
         :id => self.id,
         :creatable => @allow_creation_by_user,
-        :pulp3_support => SmartProxy.pulp_primary.pulp3_repository_type_support?(self)
+        :pulp3_support => SmartProxy.pulp_primary.pulp3_repository_type_support?(self),
+        :generic_remote_options => @generic_remote_options,
```

Comment:
alright, i think to get the translated strings correctly, we'd want this method to return a translated version of those strings, all it'd need to do is return a copy of @generic_remote_options, but looping through and returning _(option[:title]) and _(option[:description]) ,  You might make a 'translated_generic_remote_options' method to help with this